src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java

```diff
@@ -1,11 +1,101 @@
 package executor.service.service.impl;
 
 import executor.service.model.ProxyConfigHolder;
+import executor.service.model.ProxyCredentials;
+import executor.service.model.ProxyNetworkConfig;
+import executor.service.service.Provider;
 import executor.service.service.ProxySourcesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
 
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static executor.service.config.properties.PropertiesConstants.PROXY_CREDENTIALS;
+import static executor.service.config.properties.PropertiesConstants.PROXY_NETWORK_CONFIG;
+
+/**
+ * Class for reading properties as JSON from properties file.
+ *
+ * @author Oleksandr Tuleninov, Nikita Hurmaza
+ * @version 01
+ */
 public class ProxySourcesClientImpl implements ProxySourcesClient {
+
+    private static final Logger log = LoggerFactory.getLogger(ProxySourcesClientImpl.class);
+    public static final int DELAY = 1;
+
+    private final Provider jsonReader;
+
+    public ProxySourcesClientImpl(Provider jsonReader) {
+        this.jsonReader = jsonReader;
+    }
+
+    /**
+     * Get list with ProxyConfigHolder entities continuously.
+     *
+     * @return list with ProxyConfigHolder entities
+     */
     @Override
-    public ProxyConfigHolder getProxy() {
-        return null;
+    public Flux<ProxyConfigHolder> getProxies() {
+        var proxies = getListProxiesPrototype();
+        validateProxies(proxies);
+        return Flux.fromIterable(proxies)
+                .log()
+                .delayElements(Duration.ofSeconds(DELAY))
+                .repeat();
+    }
+
+    /**
+     * Check the list of ProxyConfigHolder entities.
+     *
+     * @param proxies list of ProxyConfigHolder entitie
+     */
+    private void validateProxies(List<ProxyConfigHolder> proxies) {
+        if (proxies == null || proxies.isEmpty()) {
+            log.error("List proxies cannot be null or empty");
+            throw new IllegalArgumentException("List cannot be null or empty");
+        }
+    }
```

Comment:
In Task another task is indicated. And obviously in this project there can be no such task. We don't have an HTTP client.
![image](https://github.com/GFLCourses5/worker/assets/140727960/d713e940-0aea-42cf-98d7-80a585735792)

Reply:
Actually, this check is exactly what Artem mentioned on meeting. Because if we do not validate that proxy works, scenarios will fail. Also proxies tend to be replaced

Reply:
I think that line 60 can be removed here. Because if the Proxy is null, then ExecutionService will apply the default Proxy, Taras and I found out this point in the lecture.